p2p/src/sync/mod.rs

```diff
@@ -516,15 +513,15 @@ where
                 self.requests.insert(
                     request_id,
                     PendingRequest {
-                        peer_id,
+                        _peer_id: peer_id,
                         request_type: RequestType::GetHeaders,
                         retry_count: 0,
                     },
                 );
                 peer.set_state(peer::PeerSyncState::UploadingHeaders);
                 Ok(())
             }
-            Err(e) => {
+            Err(_) => {
```

Comment:
Since you're logging anyway, why not print that error? Our errors in general implement the Display trait or at least the Debug trait.

Reply:
Fixed